rest/src/main/groovy/se/kb/libris/digi/DigitalReproductionAPI.groovy

```diff
@@ -0,0 +1,428 @@
+        
+        try {
+            String id = service.createDigitalReproduction(parse(request))
+            log.info("Created $id")
+            response.setHeader('Location', id)
+            response.setStatus(SC_CREATED)
+        }
+        catch (RequestException e) {
+            log.warn("${e.code} ${e.msg}")
+            sendError(response, e.code, e.msg)
+        }
+        catch (Exception e) {
+            log.error("Internal error: $e", e)
+            sendError(response, SC_INTERNAL_SERVER_ERROR, e.getMessage())
+        }
+    }
+    
+    static Map parse(HttpServletRequest request) {
+        if (request.getHeader('Content-Type') != JSONLD) {
+            throw badRequest("Header Content-Type must be $JSONLD")
+        }
+        
+        def electronic = readJson(request)
+        
+        //TODO
+        check(electronic, ['@type'], 'Electronic')
+        check(electronic, ['reproductionOf', '@id'], String.class)
+        check(electronic, ['production'], Map.class)
+        check(electronic, ['production', '@type'], 'Reproduction')
+        check(electronic, ['production', 'agent'], Object.class)
+        check(electronic, ['production', 'place'], Object.class)
+        check(electronic, ['production', 'year'], String.class)
+        
+        return electronic
+    }
+    
+    static void check(thing, path, expected) {
+        def actual = getAtPath(thing, path)
+        def ok = expected instanceof Class 
+                ? expected.isInstance(actual) 
+                : expected == actual
+        
+        if (!ok) {
+            def e = expected instanceof Class ? expected.getSimpleName() : expected
+            throw new RequestException(code: SC_BAD_REQUEST, msg: "Expected $e at $path, got: ${actual ?: '<MISSING>'}")
+        }
+    }
+    
+    static Map readJson(HttpServletRequest request) {
+        try {
+            mapper.readValue(request.getInputStream().getBytes(), Map)
+        } catch (JsonParseException e) {
+            throw badRequest("Bad JSON: ${e.message}")
+        }
+    }
+
+    static void sendError(HttpServletResponse response, int code, String msg) {
+        response.setStatus(code)
+        response.setHeader('Content-Type', 'application/json')
+        mapper.writeValue(response.getOutputStream(), ['code': code, 'msg' : msg ?: ''])
+    }
+    
+    //TODO
+    static String getXlAPI(HttpServletRequest request) {
+        "http://localhost:${request.getServerPort()}/"
+    }
+}
+
+@Log
+class ReproductionService {
+    private static final def DIGI = ['@id': 'https://libris.kb.se/library/DIGI']
+    private static final def DST = ['@id': 'https://libris.kb.se/library/DST']
+    private static final def FACSIMILE = ['@id': 'https://id.kb.se/term/saogf/Faksimiler']
+    private static final def ONLINE = ['@id': 'https://id.kb.se/term/rda/OnlineResource']
+    private static final def FREELY_AVAILABLE = ['@id': 'https://id.kb.se/term/freelyAvailable'] //TODO
```

Comment:
I think `https://id.kb.se/policy/freely-available` (or `free`) is what we're aiming for until something better comes along.